### ISSUE TITLE
Notice caused by undefined variable

### ISSUE THREAD

**Issue by deltafactory** [szepeviktor/fix-w3tc#470]
See $empty here:
https://github.com/szepeviktor/w3-total-cache-fixed/blob/v0.9.5.x/Cdn_Plugin.php#L774

**Comment by nigrosimone** [comment]
Thanks i fix it

**Comment by szepeviktor** [comment]
It must be a design flaw or legacy code.

**Comment by nigrosimone** [comment]
> It must be a design flaw or legacy code.

@szepeviktor  it seem legacy code.

this bug also exists in the current fix version
https://github.com/szepeviktor/w3-total-cache-fixed/blob/v0.9.5.x/Cdn_Plugin.php#L774

but all the `w3tc_attachment_url() `code doesn't exists in the latest official version:
https://plugins.trac.wordpress.org/browser/w3-total-cache/tags/0.9.5.4/Cdn_Plugin.php

maybe, the simplest fix is replace `$empty` with `''`:

```php
function w3tc_attachment_url( $url ) {
        static $allowed_files = null;

		if ( ( defined( 'WP_ADMIN' ) && $this->_config->get_boolean( 'cdn.admin.media_library' ) ) ||
			 ( $this->can_cdn() && $this->can_cdn2( '' ) ) ) {
```

`can_cdn2` expect a `$buffer` (the html of the cached page), but here has no-sense.

**Comment by nigrosimone** [comment]
Fixed into https://github.com/Furniel/W3-Total-Cache/commit/bf6f609d332725fdff56bc5019d337ddef86d84e
